src/Streamly.hs

```diff
@@ -115,45 +117,55 @@ module Streamly
     , AsyncT
     , WAsyncT
     , ParallelT
+    , mkAsync
 
     -- ** Zipping Streams
     -- $zipping
     , ZipSerialM
     , ZipAsyncM
 
-    -- * Parallel Function Application
-    -- $application
-    , (IP.|$)
-    , (IP.|&)
-    , (IP.|$.)
-    , (IP.|&.)
-    , mkAsync
+    -- * IO Streams
+    , Serial
+    , WSerial
+    , Ahead
+    , Async
+    , WAsync
+    , Parallel
+    , ZipSerial
+    , ZipAsync
 
-    -- * Merging Streams
-    -- $sum
+    -- * Re-exports
+    , Semigroup (..)
+
+    -- * Deprecated
+    -- ** Parallel Function Application
+    , (|$)
+    , (|&)
+    , (|$.)
+    , (|&.)
+
+    -- ** Merging Streams
     , serial
     , wSerial
     , ahead
     , async
     , wAsync
     , parallel
 
-    -- * Concurrency Control
-    -- $concurrency
+    -- ** Concurrency Control
     , maxThreads
     , maxBuffer
 
-    -- * Rate Limiting
-    , Rate (..)
+    -- ** Rate Limiting
+    , Rate
     , rate
     , avgRate
     , minRate
     , maxRate
     , constRate
 
-    -- * Stream Type Adapters
-    -- $adapters
-    , IsStream ()
+    -- ** Stream Type Adapters
+    , IsStream
```

Comment:
We do not want to expose the type class methods.

Reply:
In a small example I tested, it doesn't expose the typeclass methods.